### ISSUE TITLE
rust 1.8.0 on osx does not work

### ISSUE THREAD

**Issue by debris** [travis-ci/travis-ci#9128]
This issue has been observed on several pull requests to [servo/heapsize](https://github.com/servo/heapsize) library in last few days.

pr:
- https://github.com/servo/heapsize/pull/91
- https://github.com/servo/heapsize/pull/93
- https://github.com/servo/heapsize/pull/94

All of these builds fail only on osx where rust 1.8.0 is installed

<img width="976" alt="screen shot 2018-01-24 at 11 05 59" src="https://user-images.githubusercontent.com/2613714/35326331-a3737cfe-00f6-11e8-8b03-aa1434a203f4.png">

The build error always looks the same:

```
##
# Host Database
#
# localhost is used to configure the loopback interface
# when the system is booting.  Do not change this entry.
##
127.0.0.1	localhost Traviss-Mac-496.local
255.255.255.255	broadcasthost
::1             localhost
Fix WWDRCA Certificate
Unable to delete certificate matching "0950B6CD3D2F37EA246A1AAA20DFAADBD6FE1F75"security: AppleWWDRCA.cer: already in /Library/Keychains/System.keychain
$ rvm use
Using /Users/travis/.rvm/gems/ruby-2.4.2
/Users/travis/.travis/job_stages: line 436: yarn: command not found
git.checkout
0.62s$ git clone --depth=50 https://github.com/servo/heapsize.git servo/heapsize
Cloning into 'servo/heapsize'...
remote: Counting objects: 380, done.
remote: Compressing objects: 100% (194/194), done.
remote: Total 380 (delta 169), reused 351 (delta 146), pack-reused 0
Receiving objects: 100% (380/380), 67.66 KiB | 3.76 MiB/s, done.
Resolving deltas: 100% (169/169), done.
$ cd servo/heapsize
0.32s$ git fetch origin +refs/pull/93/merge:
remote: Counting objects: 6, done.
remote: Compressing objects: 100% (2/2), done.
remote: Total 6 (delta 3), reused 5 (delta 3), pack-reused 0
Unpacking objects: 100% (6/6), done.
From https://github.com/servo/heapsize
 * branch            refs/pull/93/merge -> FETCH_HEAD
$ git checkout -qf FETCH_HEAD
rustup-install
Installing Rust
36.19s$ curl -sSf https://build.travis-ci.org/files/rustup-init.sh | sh -s -- --default-toolchain=$TRAVIS_RUST_VERSION -y
info: downloading installer
info: syncing channel updates for '1.8.0-x86_64-apple-darwin'
info: latest update on 2016-04-12, rust version 1.8.0 (db2939409 2016-04-11)
info: downloading component 'rustc'
 37.1 MiB /  37.1 MiB (100 %)   5.2 MiB/s ETA:   0 s                
info: downloading component 'rust-std'
 50.3 MiB /  50.3 MiB (100 %)   9.9 MiB/s ETA:   0 s                
info: downloading component 'rust-docs'
  5.4 MiB /   5.4 MiB (100 %)   1.3 MiB/s ETA:   0 s                
info: downloading component 'cargo'
  2.5 MiB /   2.5 MiB (100 %) 882.6 KiB/s ETA:   0 s                
info: installing component 'rustc'
info: installing component 'rust-std'
info: installing component 'rust-docs'
info: installing component 'cargo'
info: default toolchain set to '1.8.0'
  1.8.0 installed - (timeout reading rustc version)
$ export PATH=$HOME/.cargo/bin:$PATH
$ rustc --version
rustc(1098,0x7fffa71e83c0) malloc: *** malloc_zone_unregister() failed for 0x7fffa71de000
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#Build-times-out-because-no-output-was-received
The build has been terminated
```

`1.8.0` used to work in the past, but it's hard to tell when it got broken. The last successful build of this library on `osx` with `1.8.0` happened on [18 Oct 2017](https://travis-ci.org/servo/heapsize/builds/288167303).

**Comment by BanzaiMan** [comment]
Does it work on your development machine?

We switched the default Mac image [at the end of November](https://blog.travis-ci.com/2017-11-21-xcode8-3-default-image-announce). Perhaps this is contributing to the issue. Perhaps try a different image: https://docs.travis-ci.com/user/reference/osx/.

**Comment by debris** [comment]
> Does it work on your development machine?

Great question! Actually, it does not. Nothing below `rustc 1.11.0` works on my machine (`macOS High Sierra 10.13.2`)

**Comment by BanzaiMan** [comment]
Try an older image (say, `xcode6.4`), but do note that older images will be retired at some not-so-distant future.

**Comment by weitjong** [comment]
Lately our project Mac CI builds also had exactly this issue of "No output has been ..." appearing in our log.  I don't know for sure when exactly it started because our Mac CI jobs have not run at all for quite some days because of the backlog). Now the backlog seems to clear and we got chance to run our jobs but hitting this. I have tried the following images: xcode9.2, xcode9.1, and xcode8.3. All got the similar output stalling issue after the jobs ran for a few minutes and stalled at exactly the same spot in our CI jobs, which makes me believe some how our jobs could have "triggered" it to happen, but I must say that this has never happened before. And interestingly, the last two of our CI jobs test build using Makefile instead of xcodebuild actually work as before. Our build script uses CMake, whether one uses Makefile generator or xcode generator should not make any difference to Travis. And yet it appears all xcode build job stalled and Makefile one was fine. again, this has never happened before, I only observed this in this latest episode.
